pydoctor/astbuilder.py

```diff
@@ -635,14 +638,17 @@ def _infer_type(expr):
     except ValueError:
         return None
     else:
-        return _annotation_for_value(value)
+        ann = _annotation_for_value(value)
```

Comment:
maybe also pass `expr` as an argument to `_annotation_for_value` so that it can `fix_missing_locations` and return a fully resolved value.

Reply:
While that would simplify `_infer_type()`, it would move the complexity to `_annotation_for_value()`, which is both more complex already and has two non-None returns versus one. Also it can be called recursively via `_annotation_for_value()`, so we'd have to pass `expr` to that as well. All in all, I think this change would be a net negative for readability.